internal/testcontext/compile.go

```diff
@@ -34,28 +35,28 @@ func (ctx *Context) Compile(pkg string) string {
 }
 
 // CompileShared compiles pkg as c-shared.
-func (ctx *Context) CompileShared(name string, pkg string) Include {
-	ctx.test.Helper()
+func (ctx *Context) CompileShared(t *testing.T, name string, pkg string) Include {
```

Comment:
I think the fix here should be to make the running in separate folders work. Let's add a TODO note about it for now.

Reply:
Can you elaborate on the folders bit, I don't quite follow. Afaik the issue here was that the test in the testcontext and the test that's calling `CompileC`.

Reply:
I mean, for compilation we should be able to use different testcontexts and folders for compilation. However, currently it's easier if the shared lib is in the same folder as the executable.

Reply:
Gotcha, I'll add that Todo.